### PR TITLE
Self-managed node groups update

### PR DESCRIPTION
**Pull request by vara-bonthu**
*Issue #185 #175 , if available:*

*Description of changes:*
- Fix for above mentioned issues
- Docs update
- IAM policies code simplified
- Allowed users to pass additional policies if required

*TEST RESULTS*
- Local end to end test is successful with example1
- E2E tests will be executed before merge

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: modules/aws-eks-managed-node-groups/iam.tf [hunk 1]
```diff
@@ -18,22 +18,8 @@ resource "aws_iam_instance_profile" "managed_ng" {
...
-resource "aws_iam_role_policy_attachment" "managed_ng_AmazonEC2ContainerRegistryReadOnly" {
-  policy_arn = "${local.policy_arn_prefix}/AmazonEC2ContainerRegistryReadOnly"
-  role       = aws_iam_role.managed_ng.name
-}
-
-resource "aws_iam_role_policy_attachment" "managed_ng_AmazonSSMManagedInstanceCore" {
-  policy_arn = "${local.policy_arn_prefix}/AmazonSSMManagedInstanceCore"
+resource "aws_iam_role_policy_attachment" "managed_ng" {
+  for_each   = local.eks_worker_policies
+  policy_arn = each.key
```
Comment (askulkarni2) [inline modules/aws-eks-managed-node-groups/iam.tf:23]: Nice

File: modules/aws-eks-managed-node-groups/locals.tf [hunk 2]
```diff
@@ -63,8 +64,15 @@ locals {
     templatefile("${path.module}/templates/userdata-${local.managed_node_group["launch_template_os"]}.tpl", local.userdata_params)
   )
 
+  eks_worker_policies = toset(concat(["${local.policy_arn_prefix}/AmazonEKSWorkerNodePolicy",
+    "${local.policy_arn_prefix}/AmazonEKS_CNI_Policy",
+    "${local.policy_arn_prefix}/AmazonEC2ContainerRegistryReadOnly",
+    "${local.policy_arn_prefix}/AmazonSSMManagedInstanceCore"],
+  local.managed_node_group["additional_iam_policies"]))
```
Comment (askulkarni2) [inline modules/aws-eks-managed-node-groups/locals.tf:71]: Minor: indent.

File: modules/aws-eks-self-managed-node-groups/locals.tf [hunk 3]
```diff
@@ -76,17 +59,23 @@ locals {
...
+  ec2_principal          = "ec2.${data.aws_partition.current.dns_suffix}"
 
-  policy_arn_prefix = "arn:aws:iam::aws:policy"
-  ec2_principal     = "ec2.${data.aws_partition.current.dns_suffix}"
+  # EKS Worker Managed Policies
+  eks_worker_policies = toset(concat(["${local.policy_arn_prefix}/AmazonEKSWorkerNodePolicy",
+    "${local.policy_arn_prefix}/AmazonEKS_CNI_Policy",
+    "${local.policy_arn_prefix}/AmazonEC2ContainerRegistryReadOnly",
+    "${local.policy_arn_prefix}/AmazonSSMManagedInstanceCore"],
+  local.self_managed_node_group["additional_iam_policies"]))
```
Comment (askulkarni2) [inline modules/aws-eks-self-managed-node-groups/locals.tf:70]: Minor: indent

**Review by askulkarni2 — COMMENTED** [review]
LGTM! Just minor indentation need to be fixed.

**Review by kcoleman731 — APPROVED** [review]
LGTM. Please see @askulkarni2 indent comments.

File: modules/aws-eks-managed-node-groups/locals.tf (hunk 2 above)
Comment (vara-bonthu) [reply modules/aws-eks-managed-node-groups/locals.tf]: Done

File: modules/aws-eks-self-managed-node-groups/locals.tf (hunk 3 above)
Comment (vara-bonthu) [reply modules/aws-eks-self-managed-node-groups/locals.tf]: Done

File: docs/node-groups.md [hunk 4]
```diff
@@ -61,51 +61,60 @@ The below example demonstrates advanced configuration options for a manged node
...
+          max_size              = "3"
+          min_size              = "1"
+          subnet_ids            = []                                # Mandatory Public or Private Subnet IDs
+          additional_tags       = {
+            ExtraTag    = "m5x-on-demand"
+            Name        = "m5x-on-demand"
+            subnet_type = "private"
+          }
+          create_worker_security_group = false
+          additional_iam_policies = []
```
Comment (awsitcloudpro) [inline docs/node-groups.md:115]: It's great to have the ability to add additional policies via input.

**Review by awsitcloudpro — COMMENTED** [review]
LGTM

**Pull request merged**